internal/generator/generator_util.go

```diff
@@ -184,12 +184,39 @@ func getMasksFromPath(path string) (directive2Masks map[string][]mask, err error
 	return directive2Masks, nil
 }
 
-func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer) error {
+func genFromSrcCode(codePath string, mapVariableName string, matchFnName string, writer io.Writer,
+	filter map[string]struct{}, override map[string][]mask) error {
 	directive2Masks, err := getMasksFromPath(codePath)
 	if err != nil {
 		return err
 	}
 
+	if filter != nil {
+		dropDirectives := []string{}
+		for d := range directive2Masks {
+			if _, found := filter[d]; !found {
+				dropDirectives = append(dropDirectives, d)
+			}
+		}
+
+		for _, d := range dropDirectives {
+			delete(directive2Masks, d)
+		}
```

Comment:
we can mutate a map while iterating through it, no need for the extra slice

```suggestion
		for d := range directive2Masks {
			if _, found := filter[d]; !found {
				delete(directive2Masks, d)
			}
		}
```

Reply:
My caution before was delete may change the order of the keys since there may be a re-hash. I just checked it is safe to delete in iteration for go map. Thanks for telling me and I will fix